node/src/Unison/Runtime/Cryptography.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# Language ScopedTypeVariables #-}
+
+module Unison.Runtime.Cryptography where
+
+import Unison.Cryptography
+import Data.ByteString (ByteString)
+import Data.ByteArray as BA
+import qualified Crypto.Random as R
+
+-- cryptonite
+-- import Crypto.Cipher.Types.Base ()
+-- import Crypto.Cipher.AES.Primitive (AES)
+import qualified Crypto.Cipher.AES as AES
+import Crypto.Cipher.Types
+import Crypto.Error
+
+-- Creates a Unison.Cryptography object specialized to use the noise protocol
+-- (http://noiseprotocol.org/noise.html).
+mkCrypto :: forall cleartext symmetricKey . (ByteArrayAccess cleartext, ByteArray cleartext, ByteArrayAccess symmetricKey, ByteArray symmetricKey) => ByteString -> Cryptography ByteString symmetricKey () () () () cleartext
```

Comment:
I looked into this and I'm not sure what to do.  It looks like `cryptonite` does not use a concrete type for the key.  Here: https://github.com/haskell-crypto/cryptonite/blob/master/Crypto/Cipher/Types/Base.hs#L54 it's constrained to be a `ByteArray` and here: https://github.com/haskell-crypto/cryptonite/blob/master/Crypto/Cipher/AES/Primitive.hs#L158 it's constrained to be a `ByteArrayAccess`.  A part of me thinks that if `cryptonite` does not constrain it, then neither should we.  Do you want to make it concrete for added type safety?